src/lib/coda_plugins/coda_plugins.ml

```diff
@@ -0,0 +1,29 @@
+open Core_kernel
+
+let coda_lib' : Coda_lib.t option ref = ref None
+
+exception Not_initializing
+
+let get_coda_lib () =
+  match !coda_lib' with Some coda -> coda | None -> raise Not_initializing
+
+let init_plugins ~logger coda plugin_paths =
+  Logger.info logger "Initializing plugins" ~module_:__MODULE__
+    ~location:__LOC__ ;
+  coda_lib' := Some coda ;
+  List.iter plugin_paths ~f:(fun path ->
+      Logger.info logger "Initializing plugin from $path" ~module_:__MODULE__
+        ~location:__LOC__
+        ~metadata:[("path", `String path)] ;
+      try
+        Dynlink.loadfile path ;
+        Logger.info logger "Plugin successfully loaded from $path"
+          ~module_:__MODULE__ ~location:__LOC__
+          ~metadata:[("path", `String path)]
+      with Dynlink.Error err ->
+        Logger.error logger "Failed to load plugin from $path: $error"
+          ~module_:__MODULE__ ~location:__LOC__
+          ~metadata:
+            [ ("path", `String path)
+            ; ("error", `String (Dynlink.error_message err)) ] ) ;
+  coda_lib' := None
```

Comment:
After the plugins are loaded, regardless of error status, `get_coda_lib` always raises?

Should the exception be named `Done_initializing plugins`?

Reply:
This error can also happen before init if a plugin gets linked into the main coda binary by mistake. Since linking order can affect this (and is hell to debug) I took the stance that it should always be an error if a plugin is linked into the coda binary itself.

I've updated this to make these errors fatal.